### PR TITLE
Add support for macOS ARM64 in MacOSProcessDataReader

### PR DESCRIPTION
**Pull request by k15tfu**
Also adds arm_thread_state64_t struct for the case when ptrauth is disabled.

Related issue: #888

Based on #915, review it first pls.

### CODE REVIEW

**Comment by leculver** [comment]
I will review once #915 is checked in and this PR doesn't contain that change too.  Thanks for working on this!

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs [hunk 1]
```diff
@@ -231,20 +239,34 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
...
                 return false;
 
-            int regSize = sizeof(x86_thread_state64_t);
+            int stateFlavor;
+            int stateCount;
+            int regSize;
+            (stateFlavor, stateCount, regSize) = Architecture switch
+            {
+                Architecture.Arm64 => (Native.ARM_THREAD_STATE64, Native.ARM_THREAD_STATE64_COUNT, sizeof(arm_thread_state64_t)),
+                _ => (Native.x86_THREAD_STATE64, Native.x86_THREAD_STATE64_COUNT, sizeof(x86_thread_state64_t))
```
Comment (leculver) [inline src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs:248]: This second switch should be `Architecture.X86` and the default case should be `_ =>throw new PlatformNotSupportedException`.

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs [hunk 2]
```diff
@@ -231,20 +239,34 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
...
                         return false;
                 }
 
-                Unsafe.As<byte, x86_thread_state64_t>(ref MemoryMarshal.GetReference(buffer.AsSpan())).CopyContext(context);
+                switch (Architecture)
+                {
+                    case Architecture.Arm64:
+                        Unsafe.As<byte, arm_thread_state64_t>(ref MemoryMarshal.GetReference(buffer.AsSpan())).CopyContext(context);
+                        break;
+                    default:
```
Comment (leculver) [inline src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs:266]: This should be `case Architecture.X86`.  In the future if more architectures are added and supported for OS X, it makes the code less error prone.

**Review by leculver — CHANGES_REQUESTED** [review]
A couple of minor issues, but otherwise it looks good when those are fixed.  Thanks!

File: src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs (hunk 1 above)
Comment (mikem8361) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs:248]: Don't you mean "Architecture.X64"? or both X86/X64?  Currently .NET Core only supports X64 and ARM64 (mono may support an X86 iOS, but I don't know for sure).

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs:248]: Ah, right.  This is also why it should be explicit and not the default case.

Comment (k15tfu) [reply src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs]: Yep, `default` is for `Architecture.Amd64`. Surely I can change it, but please also note that this style is used in other places, for example:
https://github.com/microsoft/clrmd/blob/d0409daab70c27dcf4c8b8d39cf91c11ae0773e6/src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs#L370
https://github.com/microsoft/clrmd/blob/d0409daab70c27dcf4c8b8d39cf91c11ae0773e6/src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs#L229

**Review by leculver — APPROVED** [review]
LGTM thanks!

**Pull request merged**